libvast/vast/detail/array.hpp

```diff
@@ -0,0 +1,35 @@
+/******************************************************************************
+ *                    _   _____   __________                                  *
+ *                   | | / / _ | / __/_  __/     Visibility                   *
+ *                   | |/ / __ |_\ \  / /          Across                     *
+ *                   |___/_/ |_/___/ /_/       Space and Time                 *
+ *                                                                            *
+ * This file is part of VAST. It is subject to the license terms in the       *
+ * LICENSE file found in the top-level directory of this distribution and at  *
+ * http://vast.io/license. No part of VAST, including this file, may be       *
+ * copied, modified, propagated, or distributed except according to the terms *
+ * contained in the LICENSE file.                                             *
+ ******************************************************************************/
+
+#pragma once
+
+#include <array>
+
+namespace vast::detail {
+
+namespace {
+
+template <typename T, std::size_t... Is>
+constexpr std::array<T, sizeof...(Is)>
+make_array(const T& value, std::index_sequence<Is...>) {
+  return {{(static_cast<void>(Is), value)...}};
+}
+
+} // namespace
+
+template <std::size_t N, typename T>
+constexpr std::array<T, N> make_array(const T& value) {
```

Comment:
I'm not a fan of naming this `make_array` when it doesn't behave like the `std::make_array` that was voted into C++20.

Reply:
I think I'll change the implementation to make it work like the `std::make_array`, that should be able to serve the same purpose and more.

Reply:
Turns out `std::make_array` works differently and can't be used in this case. I renamed to `generate_array`.